src/main/flight/imu.c

```diff
@@ -342,3 +342,14 @@ int16_t calculateThrottleAngleCorrection(uint8_t throttle_correction_value)
         angle = 900;
     return lrintf(throttle_correction_value * sinf(angle / (900.0f * M_PIf / 2.0f)));
 }
+
+int16_t calculateThrottleAngleCorrectionAxis(uint8_t throttle_hovering_percent, uint8_t axis){
+
+    if (ABS(anglerad[axis]) > M_PIf/2)
+        return 0; //if upside down don't increase correction
+
+    uint16_t liftOffTrust = throttle_hovering_percent * 10; //to bring from 0-100 range to 0-1000 range (thrust input is in the 1000-2000 range)
+    int16_t throttleCompensation = lrintf( liftOffTrust * cosf( ABS(anglerad[axis]) ) - liftOffTrust );
```

Comment:
yes, i forgot to change the / by *, thanks; that way it should never be negative.

I was using a small bench test with pre-calculated value and checking that result are similar; now ported as a cleanflight unit-test (is my first time, hope it is ok there, especially i'm not sure where to put the extern variable) so hopefully no more pushed mistake on my side on that math.

I didn't have fly those new fix , but we had tested with sinf() version, and actually we saw an improvement on stabilization, but quantifying it is hard!